test/uri/uri_user_info_test.cc

```diff
@@ -0,0 +1,22 @@
+
+#include <gtest/gtest.h>
+#include <sourcemeta/jsontoolkit/uri.h>
+
+TEST(URI_user_info, working_example) {
```

Comment:
Can you test user info with a URN? (if it even makes sense there?)

Reply:
Also maybe test it with a relative URI as well, just for extra coverage, which might come handy when we do our own parser

Reply:
> Can you test user info with a URN? (if it even makes sense there?)

Regarding my first reading on URNs and tags, it seems userinfo does not have to deal with.

Reply:
> lso maybe test it with a relative URI as well

Relative URI represents only path right? I don't think we could use user_info, scheme or port in this case.

Reply:
On relative URIs, exactly! You can to assert that the userinfo is empty

Reply:
> Regarding my first reading on URNs and tags, it seems userinfo does not have to deal with.

yes, I confirm! Maybe still add a test that checks that for a URN, the userinfo returns a non-value?